src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -1098,6 +1083,261 @@ public async Task<string> PayInvoiceAfterSubscriptionChangeAsync(ISubscriber sub
+        var invoiceAmount = itemsForInvoice?.Sum(i => i.Amount) ?? 0;
+        var invoiceNow = invoiceAmount >= prorateThreshold;
+        if (invoiceNow)
+        {
+            // Owes more than prorateThreshold on next invoice.
+            // Invoice them and pay now instead of waiting until next billing cycle.
+
+            string cardPaymentMethodId = null;
+            var invoiceAmountDue = upcomingPreview.StartingBalance + invoiceAmount;
+            cardPaymentMethodId = GetCardPaymentMethodId(invoiceAmountDue, customer, cardPaymentMethodId);
+
+            Stripe.Invoice invoice = null;
+            var createdInvoiceItems = new List<Stripe.InvoiceItem>();
+            Braintree.Transaction braintreeTransaction = null;
+            try
+            {
+                await CreateInvoiceItemsAsync(subscriber, itemsForInvoice, pendingInvoiceItemsDict, createdInvoiceItems);
+
+                invoice = await CreateInvoiceAsync(subscriber, cardPaymentMethodId);
+
+                var invoicePayOptions = new Stripe.InvoicePayOptions();
+                await CreateBrainTreeTransactionRequestAsync(subscriber, invoice, customer, invoicePayOptions, cardPaymentMethodId, braintreeTransaction);
+
+                await InvoicePayAsync(invoicePayOptions, invoice, paymentIntentClientSecret);
+            }
+            catch (Exception e)
+            {
+                if (braintreeTransaction != null)
+                {
+                    await _btGateway.Transaction.RefundAsync(braintreeTransaction.Id);
+                }
+
+                if (invoice != null)
+                {
+                    if (invoice.Status == "paid")
+                    {
+                        // It's apparently paid, so we need to return w/o throwing an exception
+                        return new Tuple<bool, string>(false, paymentIntentClientSecret);
+                    }
+
+                    await RestoreInvoiceItemsAsync(invoice, customer, pendingInvoiceItems);
+                }
+                else
+                {
+                    foreach (var ii in createdInvoiceItems)
+                    {
+                        await _stripeAdapter.InvoiceDeleteAsync(ii.Id);
+                    }
+                }
+
+                if (e is Stripe.StripeException strEx &&
+                    (strEx.StripeError?.Message?.Contains("cannot be used because it is not verified") ?? false))
+                {
+                    throw new GatewayException("Bank account is not yet verified.");
+                }
+
+                throw;
+            }
+        }
+
+        return new Tuple<bool, string>(invoiceNow, paymentIntentClientSecret);
+    }
+
+    private string GetCardPaymentMethodId(long invoiceAmountDue, Customer customer, string cardPaymentMethodId)
+    {
+        if (invoiceAmountDue <= 0 || customer.Metadata.ContainsKey("btCustomerId")) return cardPaymentMethodId;
```

Comment:
Wrap this return statement in a control block. See our style guide here: https://contributing.bitwarden.com/contributing/code-style/csharp#control-blocks